pkg/v3/plugin/performable.go

```diff
@@ -82,10 +83,25 @@ func (p *performables) set(outcome *ocr2keepersv3.AutomationOutcome) {
 		return random.ShuffleString(performable[i].WorkID, p.keyRandSource) < random.ShuffleString(performable[j].WorkID, p.keyRandSource)
 	})
 
+	// TODO: remove this in next version, it's a temporary fix for
+	// supporting old nodes that will limit the number of results rather than the size
+	// of the outcome
```

Comment:
Need to think more deeply, but i guess this will be a breaking change even if we remove this in the next version? 

I think the ideal way to make this non breaking is to do it with a feature flag that is then turned on at once for all nodes (via config), but I think for this specific change we may just be ok to just take a breaking risk as we don't expect more than 50 performables in a round for current traffic

Reply:
> Need to think more deeply, but i guess this will be a breaking change even if we remove this in the next version?

In case all ops will update to this version, where the nodes will produce "limited" Outcome/Observation but won't validate that, they can update to next version where we'll start producing "unlimited" Outcome/Observation as the validation will accept those

> for this specific change we may just be ok to just take a breaking risk as we don't expect more than 50 performables in a round for current traffic

OK make sense

Reply:
> they can update to next version where we'll start producing "unlimited" Outcome/Observation as the validation will accept those

Yes the validation would pass but outcome consensus would still not happen if some NOPs produce limited outcome and some with more